code/go/0chain.net/smartcontract/minersc/fees2_test.go

```diff
@@ -343,35 +345,36 @@ func testPayFees(t *testing.T, minerStakes []float64, sharderStakes [][]float64,
 		sharders = append(sharders, sharder)
 	}
 
-	populateDelegates(t, append([]*MinerNode{miner}, sharders...), minerStakes, sharderStakes)
+	populateDelegates(t, append([]*minersc.MinerNode{miner}, sharders...), minerStakes, sharderStakes)
 	_, err = ctx.InsertTrieNode(miner.GetKey(), miner)
 	require.NoError(t, err)
 	for i := 0; i < numberOfSharders; i++ {
 		_, err = ctx.InsertTrieNode(sharders[i].GetKey(), sharders[i])
 		require.NoError(t, err)
 	}
-	var allSharders = &MinerNodes{
+	var allSharders = &minersc.MinerNodes{
 		Nodes: sharders,
 	}
-	err = updateAllShardersList(ctx, allSharders)
+	err = minersc.UpdateAllShardersList(ctx, allSharders)
 	require.NoError(t, err)
 
 	// Add information only relevant to view change rounds
-	config.DevConfiguration.ViewChange = zChainYaml.viewChange
+	config.Configuration().ChainConfig = chain.NewConfigImpl(&chain.ConfigData{IsViewChangeEnabled: true})
```

Comment:
So this is the reason why we are freely exporting methods in minersc. _Unit tests should not affect production code._ We should find another way to handle this. The chain should not be used in smart contracts instead stick to using ""0chain.net/chaincore/config".

I suggest using a mock object;  `0chain/chiancore/mocks.ChainConfig.ChainConfig`. Then you can use code similar to
```go
mockChainConfig = mocks.ChainConfig{}
...
mockChainConfig.On("IsViewChangeEnabled").Return(true)
...
config.Configuration().ChainConfig = chain.NewConfigImpl(mockChainConfig)
```

Reply:
To generate mocks use
```bash
./generate_mocks.sh
```
from 0chain root.